firebase-storage/src/main/java/com/google/firebase/storage/internal/StorageReferenceUri.java

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.storage.internal;
+
+import android.net.Uri;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.firebase.emulators.EmulatedServiceSettings;
+import com.google.firebase.storage.network.NetworkRequest;
+
+/**
+ * Utility class to encapsulate the two "views" of a storage URI in a single object.
+ *
+ * @hide
```

Comment:
I think you don't need `@hide` here since it is in an internal package. The rest of the classes are a bit inconsistent here.